libs/core/src/lib/popover/popover-directive/popover.directive.ts

```diff
@@ -351,18 +357,18 @@ export class PopoverDirective implements OnInit, OnDestroy, OnChanges {
         }
     }
 
-    /** @hidden */
-    @HostListener('document:click', ['$event'])
-    clickHandler(event: MouseEvent): void {
-        if (this.containerRef &&
-            this.isOpen &&
-            this.closeOnOutsideClick &&
-            event.target !== this.elRef.nativeElement &&
-            !this.elRef.nativeElement.contains(event.target) &&
-            !this.containerRef.location.nativeElement.contains(event.target)) {
-            event.preventDefault();
-            event.stopPropagation();
-            this.close();
-        }
+    private _addListenerForOutsideClick(): void {
+         this.renderer.listen('document', 'click', (event: MouseEvent) => {
+             if (this.containerRef &&
+                 this.isOpen &&
+                 this.closeOnOutsideClick &&
+                 event.target !== this.elRef.nativeElement &&
+                 !this.elRef.nativeElement.contains(event.target) &&
+                 !this.containerRef.location.nativeElement.contains(event.target)) {
```

Comment:
Is there a possibility that:

`!this.elRef.nativeElement.contains(event.target)` - is true
`!this.containerRef.location.nativeElement.contains(event.target))` - is false

Reply:
`this.elRef.nativeElement` is for control, `this.containerRef.location.nativeElement` for body, so yes